### DISCUSSION TITLE
babel/types: Use of classProperty's "accessibility" param

### DISCUSSION THREAD

**Discussion by Arooba-git** [babel/babel#14905] [Q&A] [answered]
I have been racking my brain for quite some time now but haven't been able to deduce the purpose of this "accessibility" param when it doesn't show up in the generated code or am I missing something? 🤔 
<img width="801" alt="Screen Shot 2022-09-02 at 11 34 43 AM" src="https://user-images.githubusercontent.com/56495631/188050419-5ef29b20-b002-4de5-88f4-2a7aee480524.png">

The code I'm trying to execute:
`t.classProperty(t.identifier(`timer${setTimerNumber}`), null , timerVarType ,null , false, false, false, "protected");`

Expected node: `protected  timer1: NodeJS.Timer`
Generated node `timer1:  NodeJS.Timer` (without accessibility modifier)

Is it impossible to generate accessibility modifiers via babel types? If so, shouldn't these _excluded_ parameters be removed? 🤔

**Comment by nicolo-ribaudo** [comment, marked as answer]
We only expose properties for JS syntax and proposals, and `typeAnnotation` (because it's used both by TS and Flow), because FLow params and TypeScript params would mentally conflict. However, you can do something like this:
```js
const node = Object.assign(t.classProperty(t.identifier(`timer${setTimerNumber}`), null , timerVarType), {
  accessibility: "protected"
});
```

**Reply by Arooba-git** [reply]
Oh, let me give it a try! Thank you for the prompt response. :)

**Reply by Arooba-git** [reply]
btw could you also kindly clarify what does "excluded from builder function" imply?  (in case I would need to use other such types in future).. 
Can those params still be used for JS, in the order in which they are listed (as I have used in my code snippet)? 🤔

**Reply by nicolo-ribaudo** [reply]
It means that you cannot pass them to `t.classProperty` but that you need to define them manually (like I did above)

**Reply by Arooba-git** [reply]
Noted. thanks again!
